html/src/TDocDirective.cxx

```diff
@@ -371,9 +371,9 @@ TString TDocMacroDirective::CreateSubprocessInputFile() {
    const char* pathDelimiter = ":"; // use ":" even on windows
    TObjArray* arrDirs(macroPath.Tokenize(pathDelimiter));
    TIter iDir(arrDirs);
-   TObjString* osDir = 0;
+   TObjString* osDir = nullptr;
    macroPath = "";
-   TString filenameDirPart(gSystem->DirName(filename));
+   TString filenameDirPart = gSystem->GetDirName(filename);
```

Comment:
Isn't
```suggestion
   TString filenameDirPart( gSystem->GetDirName(filename) );
```
slightly more efficient (allow optimizing away one of the TString construction)?